testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/serviceref/ServiceRefEarTestCase.java

```diff
@@ -89,7 +88,7 @@
         // and CXF guys are not willing to add more privileged blocks into their code, thus deployments need to have
         // the following permissions (note that the wsdl.properties permission is needed by wsdl4j)
         ear.addAsManifestResource(createPermissionsXmlAsset(
-                new FilePermission(System.getProperty("java.home") + File.separator + "lib" + File.separator + "wsdl.properties", "read"),
```

Comment:
Isn't extending permissions just masking some problem (missing doPriviledged block) elsewhere?

Reply:
It really depends on what is being done. @rsearls / @asoldano any information about what happens here?

Reply:
The reason ALL_FILES is needed here is because jar,
/store/repository/org/jboss/ws/cxf/jbossws-cxf-factories/5.2.4.Final/jbossws-cxf-factories-5.2.4.Final.jar,
 is being referenced by the app, 
This is referring to a jar in the "users" maven repo.  The repo is located in
different directories for different users.  When I run this test it looks in
/home/rsearls/.m2/repository/org/jboss/ws/cxf/jbossws-cxf-factories/5.2.4.Final/jbossws-cxf-factories-5.2.4.Final.jar.
For you there will be some other root directory path.  In addition over time the
jar's version number will change so hard coding the reference as in 
 new FilePermission(System.getProperty("java.home") + File.separator + "lib" + File.separator + "wsdl.properties", "read")
will not work.   

I am not aware of FilePermission allowing wildcards in the file name other wise
we could have declared something like this 
new FilePermission("/*/repository/org/jboss/ws/cxf/jbossws-cxf-factories/*.Final/jbossws-cxf-factories-*.Final.jar")
but will not work.

Reply:
Note the bit about the user's maven repo is presumably because the server is run from an equivalent of the 'build/target' dir, i.e. it's a thin server build where the module.xml files point to maven coordinates, not to local artifacts. If it was run from a standard 'thick' installation a la 'dist/target' then the relevant paths would not be in the user's maven repo, they would be in $JBOSS_HOME/modules/system/layers/base/...

This doesn't change anything re the need for ALL_FILES; I'm just noting it because the bit about the user's maven repo makes this sound different from other ServiceLoader + security manager issues we see, but really it's another variation of the same thing.  Here the app includes a subclass of javax.xml.ws.Service, whose constructor calls javax.xml.ws.spi.Provider.provider(), which does a ServiceLoader.load() call with no privileged block.

If the javax.xml.ws.Service constructor or Provider.provider() really should be using a privileged block, we should just ignore this test when a security manager is installed (@jamezp is working up a general utility for that.) If it's wrong for those classes to use a privileged block and its right for a deployment that directly uses them to require file read permissions (a la what's already necessary in this class) then this permission change is the right thing to do.